pyiron_atomistics/atomistics/job/atomistic.py

```diff
@@ -413,6 +415,21 @@ def restart(self, job_name=None, job_type=None):
         new_ham._generic_input['structure'] = 'atoms'
         return new_ham
 
+    def check_if_file_exists(self, filename) -> None:
```

Comment:
I see that it always the pattern:

```python
self.check_if_file_exists(file_name)
new_ham.restart_file_list.append(posixpath.join(self.working_directory, file_name))
```

Doesn't it make sense to combine the two and do something like this?

```python
def ensure_file_exists_and_return(self, filename) -> str:
    file_to_return = posixpath.join(self.working_directory, filename)
    if not os.path.isfile(file_to_return):
        raise FileNotFoundError(f"File {filename} not found in working directory: {self.working_directory}")
    return file_to_return
```

And

```python
new_ham.restart_file_list.append(ensure_file_exists_and_return(file_name))
```